### PR TITLE
DTLS 1.3: Don't error out on app data before finishing handshake

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -20642,32 +20642,41 @@ int DoApplicationData(WOLFSSL* ssl, byte* input, word32* inOutIdx, int sniff)
...
-            }
-            WOLFSSL_ERROR_VERBOSE(OUT_OF_ORDER_E);
-            return OUT_OF_ORDER_E;
-        }
+#ifdef WOLFSSL_EARLY_DATA
+    if (ssl->options.tls1_3 && ssl->options.handShakeDone == 0 &&
+            ssl->options.side == WOLFSSL_SERVER_END &&
+            ssl->earlyData != no_early_data &&
+            ssl->options.clientState == CLIENT_HELLO_COMPLETE) {
+        WOLFSSL_MSG("Processing EarlyData");
```
Comment (rizlik) [inline src/internal.c:20651]: Q: Do we check later that this is indeed early data (epoch 1) and not out-of-order app data?

Comment (julek-wolfssl) [reply src/internal.c:20651]: Added explicit check.

**Comment by julek-wolfssl** [comment]
Retest this please

**Comment by dgarske** [comment]
@julek-wolfssl is this related to a Zendesk ticket? Please add a description if possible about how it was found and how to reproduce. @cconlon seems like you might have some knowledge of this since he assigned you?

**Comment by julek-wolfssl** [comment]
@dgarske  This was found by @cconlon when testing DTLS inside Java.

**Review by douzzer — APPROVED** [review]

**Pull request merged**